### DISCUSSION TITLE
Handling API Keys within the Plugin UI

### DISCUSSION THREAD

**Discussion by namithj** [aspirepress/AspireUpdate#35] [Ideas]
Currently we have the API key as a separate fields and it needs to be repeated for each host so that there can be an API key for each host.

I am thinking about using the API Key end point for dual purposes 
a) To Get an API Key
b) To Make sure we are dealing with an actual mirror

We can also rename the API endpoint to something else like Handshake or hello to make more sense in this role if required.

This is the proposed workflow

1) User enters a Host Domain

2) The Plugin tries to ping the API Key domain.

3a) The plugin receives an API Key (blank key is also fine if the host doesnt want to use an API key). We go to Step 5
3b) The plugin is unable to find the API Key endpoint.

4) We show an error message to the user and refuse to save the new host

5) If the host is approved or a failed host is removed we save the host.

6) System uses one of the added hosts randomly to ping (This can actually be finalized later if we want to prioritize or lock particular installations to a particular host etc)

**Comment by ipajen** [comment]
“User enters a Host Domain”
Instead of user enters, could it be a prefilled list of domains to choose from?  With the possibility to enter a domain.

Also Maybe api.aspirecloud.org should be default value to make it easy for end users?

**Comment by namithj** [comment]
That would make someone (in the plugins case us) a curator of mirrors, which is not something we have discussed yet.  Making it a dropdown will be user friendly though.

**Reply by harikrishnanr994** [reply]
Dropdowns are good if they are curated and properly handled. How about manually adding custom endpoints/hosts(current method) Like an advanced option? Are we limiting the hosts to 10 nos?

**Comment by namithj** [comment]
How about having a dropdown with Aspirepress and Other as options for phase 1. Selecting Other will show a text box to manually enter any host.

**Reply by namithj** [reply]
We can limit the number of hosts to 1 for Phase 1 (Aspirepress or Other )
